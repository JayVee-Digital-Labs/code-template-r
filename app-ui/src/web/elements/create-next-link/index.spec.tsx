import React from 'react'
import { mount } from 'cypress-react-unit-test'

import { createNextLink } from './'
import MockNextLink from './index.mock'

const href = '/'
const text = 'text'

describe('Creating a Web UI Next JS Link', () => {
  it('should match component html snapshot to match nextJs attributes', () => {
    const InternalLink = createNextLink(MockNextLink)
    mount(<InternalLink text={text} href={href} />)
    cy.findByText(text).toMatchSnapshot()
  })

  it('should match component image snapshot', () => {
    const InternalLink = createNextLink(MockNextLink)
    mount(<InternalLink text={text} href={href} />)
    cy.findByText(text).toMatchImageSnapshot()
  })

  it('should override default optional style', () => {
    const newStyle = {
      color: 'red',
    }

    const InternalLink = createNextLink(MockNextLink)
    mount(<InternalLink text={text} href={href} styleOptions={newStyle} />)
    cy.findByText(text).toMatchImageSnapshot()
  })

  it('should take in a optional key to match NextJs Attributes', () => {
    const key = 'key'
    const InternalLink = createNextLink(MockNextLink)

    mount(<InternalLink text={text} href={href} key={key} />)
    cy.findByText(text).toMatchSnapshot()
  })
})
