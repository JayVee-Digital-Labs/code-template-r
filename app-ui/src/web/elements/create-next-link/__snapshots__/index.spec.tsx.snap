exports[`Creating a Web UI Next JS Link > should match component html snapshot #0`] = `
<a style="color: green;">text</a>
`;

exports[`Creating a Web UI Next JS Link > should take in a optional key #0`] = `
<a style="color: green;">text</a>
`;

exports[`Creating a Web UI Next JS Link > should match component html snapshot to match nextJs attributes #0`] = `
<a style="color: green;">text</a>
`;

exports[`Creating a Web UI Next JS Link > should take in a optional key to match NextJs Attributes #0`] = `
<a style="color: green;">text</a>
`;
